src/modules/weather/weather.js

```diff
@@ -14,12 +14,16 @@ export class Weather {
 
     async start() {
         try {
+            this._spinner = new Spinner();
```

Comment:
maybe just : const spinner before try

this._spinner can cause many issues in the future